src/object.d

```diff
@@ -1751,6 +1751,59 @@ unittest
 }
 
 
+/**
+ * Convenience mixin for trivially sub-classing exceptions
+ *
+ * Even trivially sub-classing an exception involves writing boilerplate code
+ * for the constructor to correctly pass in the source file and line number the
+ * exception was thrown from. This mixin makes it easier.
+ *
+ * Example:
+ * ---
+ * class MeaCulpa: Exception
+ * {
+ *     mixin StandardExceptionConstructor;
+ * }
+ * assertThrown!MeaCulpa(enforce!MeaCulpa(false));
+ *
+ * try throw new MeaCulpa;
+ * catch (MeaCulpa e)
+ * {
+ *     assert(e.file == __FILE__);
+ *     assert(e.line == __LINE__ - 4);
+ * }
+ * ---
+ *
+ * Note that if you want to pass in special arguments to your exception, you
+ * obviously can't use this mixin and have to write the constructor manually.
+ */
+mixin template StandardExceptionConstructor()
+{
+    this(string msg, string file = __FILE__, size_t line = __LINE__)
+    {
+        super(msg, file, line);
+    }
```

Comment:
Okay. I just tested it, and it looks like if you put a ddoc comment on the constructor, then that ddoc comment will actually show up in the exception's documentation. I did not think that that was the case. So, you're going to need to document the constructor(s) in the mixin.

As for `next`, every exception in druntime declares a constructor which takes it as do most of the Phobos exceptions. It's used for exception chaining. Please declare the constructors like so

    /++
        Params:
            msg  = The message for the exception.
            file = The file where the exception occurred.
            line = The line number where the exception occurred.
            next = The previous exception in the chain of exceptions, if any.
      +/
    this(string msg, string file = __FILE__, size_t line = __LINE__, Throwable next = null) @safe pure nothrow
    {
        super(msg, file, line, next);
    }

    /++
        Params:
            msg  = The message for the exception.
            next = The previous exception in the chain of exceptions.
            file = The file where the exception occurred.
            line = The line number where the exception occurred.
      +/
    this(string msg, Throwable next, string file = __FILE__, size_t line = __LINE__) @safe pure nothrow
    {
        super(msg, file, line, next);
    }

That _is_ the standard way to do it, and the only exceptions that don't do that are ones which have alternate constructors, because they're not simply taking a message.